src/features/filter.ts

```diff
@@ -8,5 +8,18 @@ const initialState = {
 export const filterSlice = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
-});
+  reducers: {
+    setQuery: (state, action: PayloadAction<string>) => ({
+      ...state,
+      query: action.payload,
+    }),
+    setStatus: (state, action: PayloadAction<string>) => ({
+      ...state,
+      status: action.payload,
+    }),
```

Comment:
The `setStatus` action currently accepts any string as a payload. Consider enforcing or validating that the payload is a valid status type (e.g., 'all', 'active', 'completed') to prevent invalid status values from being set.